coap-core/src/main/java/com/mbed/coap/transport/udp/IDatagramSocket.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2011-2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mbed.coap.transport.udp;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+
+public interface IDatagramSocket {
+    void send(DatagramPacket p) throws IOException;
+
+    void receive(DatagramPacket p) throws IOException;
+
+    void connect(SocketAddress addr) throws SocketException;
+
+    int getLocalPort();
+
+    void setSoTimeout(int timeout) throws SocketException;
+
+    void close();
+
+    boolean isClosed();
+
+    boolean isBound();
+
+    SocketAddress getLocalSocketAddress();
+
+    void setReuseAddress(boolean on) throws SocketException;
+
+    void setSendBufferSize(int size) throws SocketException;
+
+    void setReceiveBufferSize(int size) throws SocketException;
+
+    int getSendBufferSize() throws SocketException;
+
+    int getReceiveBufferSize() throws SocketException;
+
+    boolean getReuseAddress() throws SocketException;
```

Comment:
These could be done in actual implementation. No need to expose in this interface.

Reply:
fixed.